pkg/bloomgateway/bloomgateway.go

```diff
@@ -415,10 +398,41 @@ outer:
 	g.metrics.addUnfilteredCount(numChunksUnfiltered)
 	g.metrics.addFilteredCount(len(req.Refs))
 
-	level.Debug(g.logger).Log("msg", "return filtered chunk refs", "unfiltered", numChunksUnfiltered, "filtered", len(req.Refs))
+	level.Info(logger).Log("msg", "return filtered chunk refs", "unfiltered", numChunksUnfiltered, "filtered", len(req.Refs))
 	return &logproto.FilterChunkRefResponse{ChunkRefs: req.Refs}, nil
 }
 
+// consumeTask receives v1.Output yielded from the block querier on the task's
+// result channel and stores them on the task.
+// In case the context task is done, it drains the remaining items until the
+// task is closed by the worker.
+// Once the tasks is closed, it will send the task with the results from the
+// block querier to the supplied task channel.
+func consumeTask(ctx context.Context, task Task, tasksCh chan<- Task, logger log.Logger) {
+	logger = log.With(logger, "task", task.ID)
+
+	task.responses = responsesPool.Get(len(task.series))
+	defer responsesPool.Put(task.responses)
+
+	for res := range task.resCh {
+		select {
+		case <-ctx.Done():
+			level.Debug(logger).Log("msg", "drop partial result", "fp_int", uint64(res.Fp), "fp_hex", res.Fp, "chunks_to_remove", res.Removals.Len())
```

Comment:
> In case the context task is done, it drains the remaining items until the task is closed by the worker.

By *drain* I understand removing the enqueued items for this task, but I don't see that happening here. At least, shouldn't we break the for loop over the channel or return?

Reply:
I guess we expect the context to be checked on the worker level and the resCh to be closed?